src/components/LoginPopUp.tsx

```diff
@@ -1,34 +1,112 @@
- 
-import "../Styling/LoginPopUp.css"
+import "../Styling/LoginPopUp.css";
+import "../state/authState/authContext";
+import React, { useState, useContext, useEffect } from 'react';
+import { AuthContext } from '../state/authState/authContext';
+
 interface LoginPopUpProps {
-    handleLogin: () => void;
     handleClose: () => void;
 }
 
-function LoginPopUp({ handleLogin, handleClose }: LoginPopUpProps) {
+function LoginPopUp({ handleClose }: LoginPopUpProps) {
+    const { state, dispatch } = useContext(AuthContext);
+    //const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    //const [emailError, setEmailError] = useState('');
+    const [username, setUsername] = useState('');
+    const [usernameError, setUsernameError] = useState('');
+    const [passwordError, setPasswordError] = useState('');
+    const [isFormValid, setIsFormValid] = useState(false);
+
+    // Email and password validators
+    /*
+    const validateEmail = (email: string) => {
+        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+        if (!emailRegex.test(email)) {
+            setEmailError('Invalid email format');
+        } else {
+            setEmailError('');
+        }
+    };*/
+
+    const validateUsername = (username: string) => {
+        if (username.length < 4 || username.length > 29) {
+            setUsernameError('Username must be between 3 and 30 characters');
+        } else {
+            setUsernameError('');
+        }
+    };
+
+    const validatePassword = (password: string) => {
+        if (password.length < 8 || password.length > 50) {
+            setPasswordError('Password must be between 8 and 50 characters');
+        } else {
+            setPasswordError('');
+        }
```

Comment:
Ideally these constraints would come from the API instead of being hard coded.

The constraints should be listed on the actions on `/auth`
Although, currently they are not

Not a PR block tho